src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.authtoken.jwt;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.LongSupplier;
+
+import com.google.common.base.Strings;
+import org.apache.cxf.jaxrs.json.basic.JsonMapObjectReaderWriter;
+import org.apache.cxf.rs.security.jose.jwk.JsonWebKey;
+import org.apache.cxf.rs.security.jose.jwk.KeyType;
+import org.apache.cxf.rs.security.jose.jwk.PublicKeyUse;
+import org.apache.cxf.rs.security.jose.jws.JwsUtils;
+import org.apache.cxf.rs.security.jose.jwt.JoseJwtProducer;
+import org.apache.cxf.rs.security.jose.jwt.JwtClaims;
+import org.apache.cxf.rs.security.jose.jwt.JwtToken;
+import org.apache.cxf.rs.security.jose.jwt.JwtUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.ssl.util.ExceptionUtils;
+import org.opensearch.security.support.ConfigConstants;
+
+public class JwtVendor {
+    private static final Logger logger = LogManager.getLogger(JwtVendor.class);
+
+    private static JsonMapObjectReaderWriter jsonMapReaderWriter = new JsonMapObjectReaderWriter();
+
+    private final String claimsEncryptionKey;
+    private final JsonWebKey signingKey;
+    private final JoseJwtProducer jwtProducer;
+    private final LongSupplier timeProvider;
+    private final Boolean bwcModeEnabled;
+
+    public JwtVendor(final Settings settings, final Optional<LongSupplier> timeProvider) {
+        JoseJwtProducer jwtProducer = new JoseJwtProducer();
+        try {
+            this.signingKey = createJwkFromSettings(settings);
+        } catch (Exception e) {
+            throw ExceptionUtils.createJwkCreationException(e);
+        }
+        this.jwtProducer = jwtProducer;
+        if (settings.get("encryption_key") == null) {
+            throw new IllegalArgumentException("encryption_key cannot be null");
+        } else {
+            this.claimsEncryptionKey = settings.get("encryption_key");
+        }
+        if (timeProvider.isPresent()) {
+            this.timeProvider = timeProvider.get();
+        } else {
+            this.timeProvider = () -> System.currentTimeMillis() / 1000;
+        }
+        // CS-SUPPRESS-SINGLE: RegexpSingleline get Extensions Settings
+        this.bwcModeEnabled = settings.getAsBoolean(
+            ConfigConstants.EXTENSIONS_BWC_PLUGIN_MODE,
+            ConfigConstants.EXTENSIONS_BWC_PLUGIN_MODE_DEFAULT
+        );
```

Comment:
This is enabling bwc mode on **all** usages of the JWTVendor, that doesn't seem right, it should be enabled on an individual basis, on for anomaly detection extension and off for random internet extension?

Reply:
Good catch Peter! Unless there is a separate JWT Vender for each extension, then this should not be a class variable.

FYI You can extract the security settings from the ExtensionsManager like this: https://github.com/cwperks/security/pull/6/files#diff-c4f04e2a75de01aa19f8edc6d9c3cc5596627508ff8e7f4afed97363d85b4be0R139-R148

```
Optional<ExtensionsSettings.Extension> matchingExtension = OpenSearchSecurityPlugin.GuiceHolder.getExtensionsManager()
            .lookupExtensionSettingsById(audience);
if (matchingExtension.isPresent()) {
    boolean sendBackendRoles = (boolean) matchingExtension.get().getAdditionalSettings().get(SEND_BACKEND_ROLES_SETTING);
    if (sendBackendRoles) {
        jwtClaims.setProperty("br", backendRoles);
    }
}
```

Reply:
I just made the updates:
1. I removed the incorrect constructor setup of these 2 lines
2. I setup a boolean parameter in the `createJwt()` called `roleSecurityMode`
3. In the CreateOnBehalfOfTokenAction, I made another optional API parameter called roleSecurityMode'.
4. As a friendly reminder: since we are naming it in this way, and I think it is reasonable to make it is `true` by default, and if it is specified as 'false' it will include the plain text backend roles into the claim.